### PR TITLE
CSHARP-5005: Add .NET 8 tests to Astrolabe

### PR DESCRIPTION
**Pull request by BorisDog**
Also fixes current setup failure.

### CODE REVIEW

File: integrations/dotnet/install-driver.sh [hunk 1]
```diff
@@ -38,11 +39,13 @@ else
...
 
     bash ./integrations/dotnet/dotnet-install.sh -Channel 6.0 --install-dir .dotnet --no-path
+    bash ./integrations/dotnet/dotnet-install.sh -Channel 8.0 --install-dir .dotnet --no-path
 fi
 
 # /p required to get around https://github.com/dotnet/sdk/issues/12159
-./.dotnet/dotnet build mongo-csharp-driver /p:Platform="Any CPU" # platform needs a space when building
+./.dotnet/dotnet build mongo-csharp-driver /p:Platform="Any CPU" -c Debug # platform needs a space when building
 ./.dotnet/dotnet publish mongo-csharp-driver/tests/AstrolabeWorkloadExecutor \
+    -c Debug \
```
Comment (BorisDog) [inline integrations/dotnet/install-driver.sh:48]: In .net8 `dotnet publish` looks by default for `Release` config, we assumed that the default was `Debug`.

Comment (JamesKovacs) [reply integrations/dotnet/install-driver.sh:48]: Shouldn't we be building and testing release binaries?

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Comment by blink1073** [comment]
The workload executor tests are still failing with: `AssertionError: WorkloadExecutorError: The workload executor did not write a results.json file in the expected location`

**Comment by BorisDog** [comment]
> The workload executor tests are still failing with: `AssertionError: WorkloadExecutorError: The workload executor did not write a results.json file in the expected location`

@blink1073 this is an existing failure, tracked in [CSHARP-4323](https://jira.mongodb.org/browse/CSHARP-4323)

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**